src/my_files.rs

```diff
@@ -491,46 +504,72 @@ impl MyFiles {
     ) -> Result<(), rusqlite::Error> {
         let str_filepath = file_path.to_str().unwrap();
-
-        let mut statement = self.connection_pool.prepare(
-            "INSERT INTO tidy_scores (misnamed, duplicated, unused)
-            VALUES (?1, ?2, ?3)",
-        )?;
-
+        let mut statement: rusqlite::Statement<'_>;
         let duplicated_score = match &tidy_score.duplicated {
             Some(duplicated) => !duplicated.is_empty(),
             None => false,
         };
 
-        let tidy_score_id = statement.insert(params![
-            tidy_score.misnamed,
-            duplicated_score,
-            tidy_score.unused
-        ])?;
-
-        let mut statement = self.connection_pool.prepare(
-            "UPDATE my_files
-            SET tidy_score = ?1
-            WHERE path = ?2",
+        statement = self.connection_pool.prepare(
+            "SELECT tidy_score FROM my_files
+            WHERE path = ?1",
         )?;
-        // The potential failure of this query will be handled in future work on the my_files error handling
-        let _: Result<(), _> = match statement.execute(params![tidy_score_id, str_filepath]) {
-            Ok(_) => Ok(info!("tidy_score set for file {:?}", str_filepath)),
-            Err(error) => {
-                error!(
-                    "Error setting tidy_score for file {:?}: {}",
-                    file_path, error
-                );
-                Err(error)
-            }
-        };
+        let mut current_tidy_score_id: Option<i64> = statement
+            .query_row(params![str_filepath], |row| {
+                Ok(row.get::<_, Option<i64>>(0)?)
+            })?;
+
+        // If the file already has a tidyscore attached to it, we update it
+        if current_tidy_score_id.is_some() {
+            statement = self.connection_pool.prepare(
+                "UPDATE tidy_scores
+                SET misnamed = ?1, duplicated = ?2, unused = ?3
+                WHERE id = ?4",
+            )?;
+            statement.execute(params![
+                tidy_score.misnamed,
+                duplicated_score,
+                tidy_score.unused,
+                current_tidy_score_id
+            ])?;
+        } else {
+            // If the file doesn't have a tidyscore attached to it, we create one
+            statement = self.connection_pool.prepare(
+                "INSERT INTO tidy_scores (misnamed, duplicated, unused) VALUES (?1, ?2, ?3)",
+            )?;
+            current_tidy_score_id = Some(statement.insert(params![
+                tidy_score.misnamed,
+                duplicated_score,
+                tidy_score.unused
+            ])?);
+            // And we attach it to the file
+            statement = self.connection_pool.prepare(
+                "UPDATE my_files
+                SET tidy_score = ?1
+                WHERE path = ?2",
+            )?;
+            // The potential failure of this query will be handled in future work on the my_files error handling
```

Comment:
pls add it on issues now to don't forget, i checked but didn't find it

Reply:
https://github.com/TidyBee/tidybee-agent/issues/93

This is related to this issue